crates/std_float/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+#![cfg_attr(feature = "as_crate", no_std)] // We are std!
```

Comment:
Why is this different than the `lib.rs`/`mod.rs` shenanigans in `core_simd`?

Reply:
Since it's just a single file we can just name it easily, and it doesn't cause any "dancing".

But when it's not "actually a crate", it's not allowed to set `#![no_std]`. I could probably adjust things more to be less complicated in this case (truly build as a crate and just reimport from `std`?) but I followed the same model as with `core_simd` so as to minimize any possible dependency issues in the Rust build system, as I would rather explore that as cleanup afterwards.